libraries/standard/http/include/http_client.h

```diff
@@ -550,28 +558,42 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
                               HTTPResponse_t * pResponse );
 
 /**
- * @brief Read a header from the completed response #HTTPResponse_t. This will
- * return the response header value location within #HTTPResponse_t.pBuffer.
+ * @brief Read a header from a buffer containing a complete HTTP response.
+ * This will return the location of the response header value in the
+ * #HTTPResponse_t.pBuffer buffer.
+ *
+ * @note This function should only be called on a complete HTTP response. If the
+ * request is sent through the #HTTPClient_Send function, the #HTTPResponse_t is
+ * incomplete until #HTTPClient_Send returns.
+ *
+ * @param[in] pResponse The buffer containing the completed HTTP response.
+ * @param[in] pHeaderName The header field name to read.
+ * @param[in] headerNameLen The length of the header field name in bytes.
+ * @param[out] pHeaderValueLoc This will be populated with the location of the
+ * header value in the response buffer, #HTTPResponse_t.pBuffer.
+ * @param[out] headerValueLen This will be populated with the length of the
+ * header value in bytes.
  *
- * This function should be used only a completed response. A #HTTPResponse_t is
- * not complete until #HTTPClient_Send returns.
+ * @return One of the following:
+ * - #HTTP_SUCCESS (If successful.)
+ * - #HTTP_INVALID_PARAMETER (If any provided parameters or their members are invalid.)
+ * - #HTTP_PARTIAL_RESPONSE (Part of an HTTP response was received in a partially filled response buffer.)
```

Comment:
Assuming you will add a not found error

Reply:
Yes

Reply:
If we decide to keep HTTP_PARTIAL_RESPONSE error code we will need to update the description in line 192.